### PR TITLE
Initial commit of Real Time client.  Includes authentication handshake.

### PR DESCRIPTION
**Pull request by jrkoch**

### CODE REVIEW

File: src/resources/RealTimeContext.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    } else if (event === 'delete') {
+      this.onDelete = handler;
+    } else {
+      throw new Error('You must pass either "update" or "delete" as event to on');
+    }
+
+    this.hasStartedSubscription = true;
+    this.realTimeClient.startSubscription(
+      this.entityName,
+      'CODE',
```
Comment (thzinc) [inline src/resources/RealTimeContext.js]: I think this might need to be some real data.

File: src/RealTimeClient.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      reconnectOnClose: true,
+      ...options,
+    };
+
+    // dummy object for safe calling semantics later
+    this.connection = {};
+
+    this.queuedMessages = [];
+
+    // TODO: these will (probably) be used when implementing subscriptions
```
Comment (kennethito) [inline src/RealTimeClient.js]: This still todo?

File: src/resources/RealTimeContext.js (hunk 1 above)
Comment (jrkoch) [reply src/resources/RealTimeContext.js]: I think you might be right... I'll resolve and commit the fix.  Thanks.

File: src/RealTimeClient.js (hunk 2 above)
Comment (jrkoch) [reply src/RealTimeClient.js]: Thanks.  Subscriptions will be implemented in an upcoming story.  I'll change this comment (and the other TODO) to mention that.

File: src/resources/RealTimeContext.js (hunk 1 above)
Comment (jrkoch) [reply src/resources/RealTimeContext.js]: This is now fixed

**Comment by jrkoch** [comment]
Unit tests and examples added, missing customer code bug fixed.  This should be good to go, let me know if you have any further feedback.

**Pull request merged**